src/persist/src/indexed/mod.rs

```diff
@@ -288,22 +283,16 @@ impl<K: Data, V: Data, U: Buffer, L: Blob> Indexed<K, V, U, L> {
             (k1, v1, t1).cmp(&(k2, v2, t2))
         });
 
-        // ...writing that snapshot's data into trace...
+        // ...and atomically swapping that snapshot's data into trace.
         let batch = BlobTraceBatch {
             id: id,
             desc: desc.clone(),
             updates,
         };
-        self.append_trace(key, batch)?;
-        let future = self
-            .futures
-            .get_mut(&id)
-            .ok_or_else(|| Error::from(format!("never registered: {:?}", id)))?;
-
-        // .. and removing the data from future once that's successful.
-        future.truncate(desc.upper().clone())
+        self.append_trace(key, batch)
 
-        // TODO: Incrementally compact trace.
+        // TODO: This is a good point to compact trace. The data that's been
```

Comment:
I think the second sentence is referring to future now and not trace? as in "The data that's been moved [from future] is still there but now irrelevant" I'm not sure if its a good time to compact future or trace or both at this point -- i think both?

Reply:
Done